### ISSUE TITLE
Unable to display control overlayed on Android

### ISSUE THREAD

**Issue by stefanolson** [unoplatform/uno#4836]
## Current behavior

Trying to display a dialog in an overlayed control on android results in the exception: "Java.Lang.IllegalStateException: 'The specified child already has a parent. You must call removeView() on the child's parent first.'"

## Expected behavior

The control should appear as it does on UWP.

## How to reproduce it (as minimally and precisely as possible)

Enclosed: [UnoApp.zip](https://github.com/unoplatform/uno/files/5772721/UnoApp.zip)
Run app on Android target. Click on open button. Error occurs.

## Workaround

None.

## Environment

Nuget Package:

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [X] Visual Studio 2019 (version: 16.8.3 )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by jeromelaban** [comment]
Thanks for the report.

The `ItemsControl` and it interactions with Android may using caching that prevent views from being reused this way. This may not be a correct behavior from Uno, yet it seems that you're simulating a Popup and Dialogs system. 

The WinUI popups system (`Popup`, `Flyout`, and the `ContentDialog` class) offers similar properties. Is there a reason for creating your own system ?

**Comment by stefanolson** [comment]
Yes, that's exactly what I'm simulating. I have multiple levels/layers of dialogs and ContentDialog doesn't allow multiple dialogs at the same time. pretty sure the same with Popup. Not so sure about Flyout but I assume it would be the same.  I built the system so long ago I've probably forgotten some of the other reasons. It was done when I was building both a WPF and UWP version. UWP version got canned but kept using the system. Works really well (in WPF)

Let me know if you need any further details.

**Comment by jeromelaban** [comment]
Understood. A workaround at this point would be to create a control that manipulates the Grid children explicitly, without going through an `ItemsControl`. 

Also, UWP supports multiple popups at the same time (even if `ContentDialog` does not), so I'm wondering since your popups overlap if you could use the `Popup` system instead.

**Comment by stefanolson** [comment]
Thanks, yeah trying manually adding Grid entries now. Works so far.  I think I'll stick with that because it's working fine in WPF and don't want to unnecessary change anything that doesn't need to be changed :-)

**Comment by stefanolson** [comment]
I think that Popups wouldn't work because I basically have two levels of these dialogs. Sometimes there are dialogs that need to appear at the front and so I've got a different grid displaying those dialogs. You could potentially solve it with pop-ups if setting the Z index would work but it sounds more complex the workaround you have suggested.

**Comment by MartinZikmund** [comment]
Only relevant to native Android, works correctly in Skia
